Mathlib/Algebra/Polynomial/Factors.lean

```diff
@@ -0,0 +1,296 @@
+/-
+Copyright (c) 2025 Thomas Browning. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Thomas Browning
+-/
+import Mathlib.Algebra.Polynomial.FieldDivision
+
+/-!
+# Split polynomials
+
+A polynomial `f : R[X]` factors if it is a product of constant and monic linear polynomials.
+
+## Main definitions
+
+* `Polynomial.Factors f`: A predicate on a polynomial `f` saying that `f` is a product of
+  constant and monic linear polynomials.
+
+## TODO
+
+- Redefine `Splits` in terms of `Factors` and then deprecate `Splits`.
+
+-/
+
+variable {R : Type*}
+
+namespace Polynomial
+
+section Semiring
+
+variable [Semiring R]
+
+/-- A polynomial `Factors` if it is a product of constant and monic linear polynomials.
+This will eventually replace `Polynomial.Splits`. -/
+def Factors (f : R[X]) : Prop := f ∈ Submonoid.closure ({C a | a : R} ∪ {X + C a | a : R})
```

Comment:
This is a pretty complicated definition. Would this be any nicer as an inductive? i.e. `C a` factors, `X + C a` factors, and the product of two factorable polynomials factors.

Reply:
It would, so disregard my suggestion :) 

Perhaps the correct thing to do is to make a specialization of `Submonoid.closure_induction` that separates out the cases of constant and linear polynomials.